packages/eas-cli/src/commands/update/index.ts

```diff
@@ -600,15 +630,67 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), exp];
+  } catch (error: any) {
```

Comment:
catching all errors assumes that they are all runtime version errors vs other classes of errors (code errors, other errors). There doesn't seem to be a very good way of detecting just the class of error though other than error message introspection (which I don't recommend), so I think this is okay.

The only thing we risk is that it is already configured and `transformRuntimeVersions` throws another error and we clobber their configuration accidentally.

Reply:
Only error that `transformRuntimeVersions` can throw is a one defined by us via `nullthrows` (it's a very basic method), so I think that is safe to use the generic catch here, and assume that every error is a missing runtime version.